src/emu/video/rgbgen.h

```diff
@@ -151,16 +151,20 @@ class rgbaint_t
 
 	inline void shl(const rgbaint_t& shift)
 	{
-		m_a <<= shift.m_a;
-		m_r <<= shift.m_r;
-		m_g <<= shift.m_g;
-		m_b <<= shift.m_b;
+		m_a = u32(shift.m_a) > 31 ? 0 : m_a << shift.m_a;
+		m_r = u32(shift.m_r) > 31 ? 0 : m_r << shift.m_r;
+		m_g = u32(shift.m_g) > 31 ? 0 : m_g << shift.m_g;
+		m_b = u32(shift.m_b) > 31 ? 0 : m_b << shift.m_b;
```

Comment:
Please use parentheses around arguments to the ternary conditional operator, to make things clearer, like:
```c++
		m_a = (u32(shift.m_a) > 31) ? 0 : (m_a << shift.m_a);
		m_r = (u32(shift.m_r) > 31) ? 0 : (m_r << shift.m_r);
		m_g = (u32(shift.m_g) > 31) ? 0 : (m_g << shift.m_g);
		m_b = (u32(shift.m_b) > 31) ? 0 : (m_b << shift.m_b);
```

(And in other places as well.)